### PR TITLE
Add notice for licenses of dependencies

### PR DESCRIPTION
**Pull request by timkpaine**

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,697 @@
+CSP
+Copyright 2023-2024 Point72, L.P.
+
+
+
+This product includes software from https://github.com/abseil/abseil-cpp (Apache 2.0)
+ * Copyright (c) The abseil authors
+
+This product includes software from https://github.com/google/brotli/ (BSD 3-clause)
```
Comment (ngoldbaum) [inline NOTICE:9]: Brotli is MIT https://github.com/google/brotli/blob/master/LICENSE

**Review by ngoldbaum — COMMENTED** [review]
Just one comment. Also for static linking do we need to worry about licenses of dependencies pulled in by the direct dependencies listed in this file? I know the build pulls in boost, for example.

File: NOTICE (hunk 1 above)
Comment (timkpaine) [reply NOTICE]: doing it a better way now

**Comment by timkpaine** [comment]
> Just one comment. Also for static linking do we need to worry about licenses of dependencies pulled in by the direct dependencies listed in this file? I know the build pulls in boost, for example.

Yeah i tweaked how to generate so we can pull in everything, its a big file but seems fine

**Comment by timkpaine** [comment]
also added a quick tweak for https://github.com/Point72/csp/issues/48 while i was modifying the makefile

File: Makefile [hunk 2]
```diff
@@ -114,7 +122,7 @@ major:
 ########
 # DIST #
 ########
-.PHONE: dist-py dist-py-sdist dist-py-wheel dist-py-cibw dist-check dist publish-py publish
+.PHONY: dist-py dist-py-sdist dist-py-wheel dist-py-cibw dist-check dist publish-py publish
```
Comment (ngoldbaum) [inline Makefile:125]: oops, nice catch

**Review by ngoldbaum — APPROVED** [review]
Looks good to me now, makes sense to automate this.

**Pull request merged**